src/bin/ion/commands/generate/templates/java/class.templ

```diff
@@ -111,12 +125,18 @@ import java.io.IOException;
      *
      * This method does not close the writer after writing is complete.
      * The caller is responsible for closing the stream associated with the writer.
+     * This method skips writing a field when it's null.
      */
     public void writeTo(IonWriter writer) throws IOException {
          {# Writes `Structure` class with multiple fields based on `field.name` as an Ion struct #}
          writer.stepIn(IonType.STRUCT);
          {% for field_name, field_val in struct_info["fields"] %}
              {% set field_value = field_val.0 | fully_qualified_type_name %}
+             {% set field_occurrence = field_val.1 %}
+             {% if field_occurrence == "Optional" %}
```

Comment:
The more I think about it, the more it seems like this is a good case for a [tester](https://keats.github.io/tera/docs/#tests), so that we can just write `{% if field_val is optional_field %}` or something like that. Doesn't need to be done in this PR, though.